### PR TITLE
Source LinkedIn Ads: move to next stage

### LINKED ISSUE #14803: LInkedin Ads: add breakdown by conversion name
## Tell us about the problem you're trying to solve
There are breakdowns by campaign and creative (Ad Analytics by Campaign
Ad Analytics by Creative)
but no breakdowns by conversion name. 
So we can count how many conversions were made by campaign or creative,
but we can't see which conversions exactly happened. It is crucial for every marketing department
to know not only total number of conversions, but which conversions exactly happened.

## Describe the solution you’d like
Linkedin Ads API allows adding up to 3 breakdowns(pivots)
For example, pivot: ''CONVERSION,CAMPAIGN" instead of  pivot: ''CAMPAIGN"
API request example - https://www.postman.com/linkedin-developer-apis/workspace/linkedin-marketing-solutions/example/16069442-8677edd3-860e-477b-874e-89bccba28f58

So I guess it could be a great idea to have additional streams with more granular pivots 
like we have in Facebook Ads connector.

## Describe the alternative you’ve considered or used
We are going to load this data via a Python script for now.

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolving:
- https://github.com/airbytehq/airbyte/issues/28215

Resolve https://github.com/airbytehq/airbyte/issues/14803

## How

- use latest CDK (0.50.2)
- decrease checkpoint interval to 100
- add custom ad report option:
image

## Recommended reading order
1. `setup.py`
2. `streams.py`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-linkedin-ads test report (commit [`035a2870e2`](https://github.com/airbytehq/airbyte/pull/29252/commits/035a2870e2ed91ef49b2dd8003546226c84838ae)) - ❌

⏲️  Total pipeline duration: 26mn41s 

| Step                                                                       | Result   |
|:---------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-linkedin-ads/metadata.yaml | ✅        |
| Connector version semver check                                             | ✅        |
| Connector version increment check                                          | ✅        |
| QA checks                                                                  | ✅        |
| Code format checks                                                         | ✅        |
| Connector package install                                                  | ✅        |
| Build source-linkedin-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                 | ✅        |
| Acceptance tests                                                           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-likedin-ads-move-next-stage/1692013233/035a2870e2ed91ef49b2dd8003546226c84838ae/source-linkedin-ads/0.5.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:035a2870e2ed91ef49b2dd8003546226c84838ae)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-linkedin-ads test
```

**Comment by octavia-squidington-iii** [comment]
##  source-linkedin-ads test report (commit [`035a2870e2`](https://github.com/airbytehq/airbyte/pull/29252/commits/035a2870e2ed91ef49b2dd8003546226c84838ae)) - ❌

⏲️  Total pipeline duration: 27mn32s 

| Step                                                                       | Result   |
|:---------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-linkedin-ads/metadata.yaml | ✅        |
| Connector version semver check                                             | ✅        |
| Connector version increment check                                          | ✅        |
| QA checks                                                                  | ✅        |
| Code format checks                                                         | ✅        |
| Connector package install                                                  | ✅        |
| Build source-linkedin-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                 | ✅        |
| Acceptance tests                                                           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-likedin-ads-move-next-stage/1692014928/035a2870e2ed91ef49b2dd8003546226c84838ae/source-linkedin-ads/0.5.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:035a2870e2ed91ef49b2dd8003546226c84838ae)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-linkedin-ads test
```

**Comment by octavia-squidington-iii** [comment]
##  source-linkedin-ads test report (commit [`479868ae8f`](https://github.com/airbytehq/airbyte/pull/29252/commits/479868ae8f8f71117a73c312b4b7a35087824b69)) - ✅

⏲️  Total pipeline duration: 15mn31s 

| Step                                                                       | Result   |
|:---------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-linkedin-ads/metadata.yaml | ✅        |
| Connector version semver check                                             | ✅        |
| Connector version increment check                                          | ✅        |
| QA checks                                                                  | ✅        |
| Code format checks                                                         | ✅        |
| Connector package install                                                  | ✅        |
| Build source-linkedin-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                 | ✅        |
| Acceptance tests                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-likedin-ads-move-next-stage/1692021295/479868ae8f8f71117a73c312b4b7a35087824b69/source-linkedin-ads/0.5.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:479868ae8f8f71117a73c312b4b7a35087824b69)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-linkedin-ads test
```

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/source.py [hunk 1]
```diff
@@ -84,22 +86,47 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
...
+            )
+            streams.append(stream)
+
+        return streams
+
+    def _validate_ad_analytics_reports(self, config: Mapping[str, Any]) -> None:
+        if len(config.get("ad_analytics_reports", [])) and len(config.get("ad_analytics_reports")) != set(
+            x["name"] for x in config.get("ad_analytics_reports")
+        ):
+            report_names = [x["name"] for x in config.get("ad_analytics_reports")]
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/source.py:130]:
```suggestion
        report_names = [x["name"] for x in config.get("ad_analytics_reports", [])]
        if len(report_names) != len(set(report_names)):
```

File: airbyte-integrations/connectors/source-linkedin-ads/setup.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 from setuptools import find_packages, setup
 
 MAIN_REQUIREMENTS = [
-    "airbyte-cdk~=0.37",
+    "airbyte-cdk~=0.50",
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-linkedin-ads/setup.py:9]:
```suggestion
    "airbyte-cdk",
```

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/streams.py [hunk 3]
```diff
@@ -367,10 +368,24 @@ class LinkedInAdsAnalyticsStream(IncrementalLinkedinAdsStream, ABC):
     def get_json_schema(self) -> Mapping[str, Any]:
         return ResourceSchemaLoader(package_name_from_class(self.__class__)).get_schema("ad_analytics")
 
+    def __init__(self, name: str = None, pivot_by: str = None, time_granularity: str = None, **kwargs):
+        self._new_class_name = name
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/streams.py:372]:
```suggestion
        self.user_stream_name = name
```

**Review by davydov-d — APPROVED** [review]
minor comments, otherwise looks good

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/source.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/source.py]: agreed

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/streams.py (hunk 3 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/streams.py]: renamed

**Comment by octavia-squidington-iii** [comment]
##  source-linkedin-ads test report (commit [`6da1802d24`](https://github.com/airbytehq/airbyte/pull/29252/commits/6da1802d24ca3887661b8136ffab5c6134e7b701)) - ✅

⏲️  Total pipeline duration: 34mn54s 

| Step                                                                       | Result   |
|:---------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-linkedin-ads/metadata.yaml | ✅        |
| Connector version semver check                                             | ✅        |
| Connector version increment check                                          | ✅        |
| QA checks                                                                  | ✅        |
| Code format checks                                                         | ✅        |
| Connector package install                                                  | ✅        |
| Build source-linkedin-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                 | ✅        |
| Acceptance tests                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-likedin-ads-move-next-stage/1692024099/6da1802d24ca3887661b8136ffab5c6134e7b701/source-linkedin-ads/0.5.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:6da1802d24ca3887661b8136ffab5c6134e7b701)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-linkedin-ads test
```

**Pull request merged**